src/components/shared/Validations.tsx

```diff
@@ -28,8 +33,30 @@ const numberValidationFormatters: Record<string, (value: unknown) => string> = {
   minLength: value => `>= ${value} characters`,
   maximum: value => `<= ${value}`,
   exclusiveMaximum: value => `< ${value}`,
-  maxItems: value => `< ${value} items`,
-  maxLength: value => `< ${value} characters`,
+  maxItems: value => `<= ${value} items`,
+  maxLength: value => `<= ${value} characters`,
+};
+
+const validationsFormatter = (name: string) => (value: unknown[] | unknown): ValidationFormat | null => {
```

Comment:
An `unknown[] | unknown` is still `unknown`, although I can see your point that it increases clarity somewhat. 

We shouldn't be using this many `unknown`s to begin with, but that's not your fault, that's how validations are defined unfortunately.

Reply:
Also I'd consider calling the function `createValidationsFormatter` as it's a HOF. It isn't the formatter itself, rather it *creates* the function that will be the formatter.